problems.rb

```diff
@@ -0,0 +1,85 @@
+# Your method should produce a single line of output (may wrap with long lists).
+#
+def print_array(int_array)
+  list = []
+  int_array.each do |int|
+    list << int
+  end
+  return list
+end
+
+# Problem #2
+# Write a method named stretch that accepts an array of integers as a parameter and returns a new array twice as large as the original, where every integer from the original array is replaced with a pair of integers, each half the original.
+# If a number in the original array is odd, then the first number in the new pair should be one higher than the second so that the sum equals the original number.
+# For example, if a variable named list refers to an array storing the values [18, 7, 4, 24, 11], the call of stretch(list) should return a new array containing [9, 9, 4, 3, 2, 2, 12, 12, 6, 5]. (The number 18 is stretched into the pair 9, 9, the number 7 is stretched into 4, 3, the number 4 is stretched into 2, 2, the number 24 is stretched into 12, 12 and the number 11 is stretched into 6, 5.)
+#
+def test_stretch
+  ary = [18, 7, 4, 24, 11]
+  if stretch(ary) == [9, 9, 4, 3, 2, 2, 12, 12, 6, 5]
+    return true
+  else
+    return false
+  end
+end
+
+def stretch(int_array)
+  list = []
+  int_array.each do |i|
+    if i%2 == 0 #if even
+      2.times do
+        list << (i/2)
+      end
+    else #if odd
+      list << (1 + i/2) #Ruby integers rounds down
+      list << (i/2)
+    end
+  end
+  return list
+end
+
+
+# Problem #3
+# Write a method named numUnique that accepts a sorted array of integers as a parameter and utilizes a hash to calculate and return the number of unique values in the array. The array is guaranteed to be in sorted order, which means that duplicates will be grouped together.
+# For example, if a variable called list stores the following values: list = [5, 7, 7, 7, 8, 22, 22, 23, 31, 35, 35, 40, 40, 40, 41] then the call of numUnique(list) should return 9 because this list has 9 unique values (5, 7, 8, 22, 23, 31, 35, 40 and 41).
+# It is possible that the list might not have any duplicates. For example if list instead stored this sequence of values: list = [1, 2, 11, 17, 19, 20, 23, 24, 25, 26, 31, 34, 37, 40, 41] Then a call on the method would return 15 because this list contains 15 different values.
+# If passed an empty list, your method should return 0.
+
+def test_num_unique_both
+  list1 = [5, 7, 7, 7, 8, 22, 22, 23, 31, 35, 35, 40, 40, 40, 41]
+  list2 = [1, 2, 11, 17, 19, 20, 23, 24, 25, 26, 31, 34, 37, 40, 41]
+  list3 = [0, 0, 0]
+  if num_unique(list1) == 9 && num_unique(list2) == 15 && num_unique([]) == 0 && num_unique(list3) == 1 && num_unique_2(list1) == 9 && num_unique_2(list2) == 15 && num_unique_2([]) == 0 && num_unique_2(list3) == 1
+    return true
+  else
+    return false
+  end
+end
+
+def num_unique(int_array)
+  num_hash = {}
+  int_array.each do |i|
+      num_hash[i] = "awesome"
+  end
+  puts num_hash
+  return num_hash.length
+end
```

Comment:
This works, but doesn't utilize the function of a hash, instead just uses it like a fixnum. See comments that I left for Heather's PR which has a similar solution.